src/main/java/net/minecraftforge/fml/config/ConfigLoadingException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2020.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.fml.config;
+
+public class ConfigLoadingException extends RuntimeException
+{
+    private final ModConfig config;
+
+    public ConfigLoadingException(ModConfig config, Exception cause)
+    {
+        super(cause);
+        this.config = config;
+    }
+
+    @Override
+    public String getMessage()
+    {
+        return "Failed loading config file " + config.getFullPath().toString() + " with type " + config.getType() + " for modid " + config.getModId();
```

Comment:
This seems weird, why not just print pass the message as a super argument?

Reply:
I believe that is because I looked at `LoadingFailedException` as a reference for how exceptions and their messages were constructed. Will change.